components/sbm-core/src/main/java/org/springframework/sbm/project/parser/MavenProjectParser.java

```diff
@@ -63,25 +60,15 @@ public class MavenProjectParser {
     private final ResourceParser resourceParser;
     private final RewriteMavenParser mavenParser;
     private final MavenArtifactDownloader artifactDownloader;
-//    private final JavaParser.Builder<?, ?> javaParserBuilder;
     private final ApplicationEventPublisher eventPublisher;
     private final JavaProvenanceMarkerFactory javaProvenanceMarkerFactory;
 
     private final JavaParser javaParser;
-//    public MavenProjectParser(ResourceParser resourceParser,
-//                              MavenArtifactDownloader artifactDownloader,
-//                              MavenParser.Builder mavenParserBuilder,
-//                              JavaParser.Builder<?, ?> javaParserBuilder,
-//                              ApplicationEventPublisher eventPublisher, JavaProvenanceMarkerFactory javaProvenanceMarkerFactory, ExecutionContext ctx) {
-//        this.resourceParser = resourceParser;
-//        this.mavenParser = mavenParserBuilder.build();
-//        this.artifactDownloader = artifactDownloader;
-////        this.javaParserBuilder = javaParserBuilder;
-//        this.eventPublisher = eventPublisher;
-//        this.javaProvenanceMarkerFactory = javaProvenanceMarkerFactory;
-//    }
 
     public List<SourceFile> parse(Path projectDirectory, List<Resource> resources) {
+
+        injectMavenConfigIntoSystemProperties(resources);
```

Comment:
Could we move `injectMavenConfigIntoSystemProperties()` and `getMavenConfigs()` into the `MavenConfigParser` and maybe find a better name as it does more than just parsing `.mvn/maven.config`. 
Maybe `MavenConfigHandler` or maybe you have a better name?
In test it would become easier, as we'd only need to mock this component then (currently we'd need to spy on methods or use integration tests) because the concern is split into two classes.